File: Compare_N_Contents_Two_Strings/Compare.js
/*
    Author : Pakshal Shashikant Jain
    Date : 15/09/2023
    Program : Write a program which 2 strings from user and check whether first
              N contents of two strings are equal or not. (Implement strcmp()
              function).

              Input : "Marvellous Infosystems"
                      "Marvellous Logic Building"
                      10
              Output : TRUE
*/
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Function To Compare First N Contents of Entered Strings
function StrNCmpX(arr, brr, icnt) {
    var i = 0;
    var j = 0;
    if ((arr == null) && (brr == null)) {
        return false;
    }
    //Logic To Compare
    while (((i != arr.length) && (j != brr.length)) && icnt != 0) {
        if (arr[i] != brr[i]) {
            break;
        }
        i++;
        j++;
        icnt--;
    }
    if (arr[i] != brr[i]) {
        return false;
    }
    else {
        return true;
    }
}
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Main Function
function main() {
    var size = 7;
    var str = "Pakshal Jain";
    var str2 = "Pakuhal Programmer";
    var bret = false;
    var arr;
    var brr;
    arr = Array.from(str);
    brr = Array.from(str2);
    console.log("Jay Ganesh....");
    //Call To StrNCmpX Function
    bret = StrNCmpX(arr, brr, size);
    //Handeling Condition if bret is true
    if (bret == true) {
        console.log("First N Contents of String are Equal");
    }
    //Handeling Condition if bret is false
    else {
        console.log("First N Contents of String are Not Equal");
    }
}
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Call To Main function
main();

File: Compare_N_Contents_Two_Strings/Compare.java
/*
    Author : Pakshal Shashikant Jain
    Date : 15/09/2023
    Program : Write a program which 2 strings from user and check whether first
              N contents of two strings are equal or not. (Implement strcmp()
              function).

              Input : "Marvellous Infosystems"
                      "Marvellous Logic Building"
                      10
              Output : TRUE
*/

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
import java.lang.*;
import java.util.*;

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class StringCmp
{
    //Characteristics of Class
    char arr[];
    char brr[];
    String str;
    String str2;
    int icnt;

    //Constructor of Class
    StringCmp()
    {
        this.arr = null;
        this.brr = null;
        this.str = " ";
        this.str2 = " ";
        this.icnt = 0;
    }

    //Behaviour of Class TO Take Input from user
    void Accept()
    {
        //Creating object of Scanner Class To Take Input from user
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter First String");
        this.str = sobj.nextLine();

        System.out.println("Enter Second String");
        this.str2 = sobj.nextLine();

        System.out.println("Enter One size");
        this.icnt = sobj.nextInt();

        this.arr = new char[this.str.length()];
        this.brr = new char[this.str2.length()];

        this.arr = this.str.toCharArray();
        this.brr = this.str2.toCharArray();
    }

    //Behaviour of class To Compare
    boolean StrNCmpX()
    {
        int i = 0;
        int j = 0;

        if((this.arr == null)&&(this.brr == null))
        {
            return false;
        }        

        //Logic To Compare
        while(((i != this.arr.length)&&(j != this.brr.length))&&(icnt != 0))
        {
            if(this.arr[i] != this.brr[j])
            {
                break;
            }
            i++;
            j++;
            icnt--;
        }
        if(this.arr[i] != this.brr[j])
        {
            return false;
        }
        else
        {
            return true;
        }
    }
}//End of Class

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Class
class Compare
{
    //Main Function
    public static void main(String args[])
    {
        boolean bret = false;

        System.out.println("Jay ganesh....");

        //Creating object of StringCmp Class
        StringCmp sobj = new StringCmp();

        //Call To Behaviours of Class
        sobj.Accept();
        bret = sobj.StrNCmpX();

        //Handeling Conditions According To Output
        if(bret == true)
        {
            System.out.println("First N Contents Of Both String are Equal");
        }
        else
        {
            System.out.println("First N Contents Of Both String are Not Equal");
        }
    }
}